### PR TITLE
WT-9971 Create a standard implementation of a pareto distribution function and utilize it in test format

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by luke-pearson** [comment]
This is blocked by a config.sh change introduced by WT-10793

File: test/utility/misc.c [hunk 1]
```diff
@@ -601,6 +603,34 @@ testutil_time_us(WT_SESSION *session)
...
 
+/*
+ * testutil_pareto --
+ *     Given a random value generated by __wt_random, a key range and a skew percentage generate a
+ *     pareto distributed value.
+ */
+uint32_t
+testutil_pareto(uint32_t rand, uint64_t min_key, uint64_t max_key, u_int skew)
+{
+    double S1, S2, U;
```
Comment (luke-pearson) [inline test/utility/misc.c:614]: I'm not a huge fan of these variable names but I don't know what they stand for either.

File: test/ctest_helpers.cmake [hunk 2]
```diff
@@ -116,6 +116,8 @@ function(create_test_executable target)
         )
     endif()
 
+    target_link_libraries(${target} m)
```
Comment (luke-pearson) [inline test/ctest_helpers.cmake:119]: This is an annoying side effect, from now on every test compiled under the function `create_test_executable` will be linked to the math library. A workaround could be implementing pow directly.

File: test/format/ops.c [hunk 3]
```diff
@@ -1130,6 +1130,8 @@ ops(void *arg)
         if (table->type != ROW && !table->mirror)
             WT_ORDERED_READ(max_rows, table->rows_current);
         tinfo->keyno = mmrand(&tinfo->data_rnd, 1, (u_int)max_rows);
+        if (TV(OPS_PARETO))
+            tinfo->keyno = testutil_pareto(tinfo->keyno, 1, (u_int)max_rows, 20);
```
Comment (luke-pearson) [inline test/format/ops.c:1134]: Not a fan of the hardcoded 20 but don't have a read on what value would be good there, @ddanderson.

File: test/utility/misc.c [hunk 4]
```diff
@@ -601,6 +603,34 @@ testutil_time_us(WT_SESSION *session)
     return ((uint64_t)ts.tv_sec * WT_MILLION + (uint64_t)ts.tv_nsec / WT_THOUSAND);
 }
```
Comment (ddanderson) [inline test/utility/misc.c:605]: BTW, workgen also has a pareto function (`pareto_calculation`) that should probably be changed to use this.

**Review by ddanderson — COMMENTED** [review]
The existing pareto code in wtperf is sets the rval value to be an offset, not a key, so if that is used, that should be clearly noted and the caller needs to adjust it as needed.  The `pareto_calculation` in workgen could be a clearer starting point?

This is gnarley stuff, happy to talk about it.

File: test/format/ops.c [hunk 5]
```diff
@@ -1130,6 +1130,8 @@ ops(void *arg)
         if (table->type != ROW && !table->mirror)
             WT_ORDERED_READ(max_rows, table->rows_current);
         tinfo->keyno = mmrand(&tinfo->data_rnd, 1, (u_int)max_rows);
+        if (TV(OPS_PARETO))
+            tinfo->keyno = testutil_pareto((uint32_t)tinfo->keyno, 1, (u_int)max_rows, 20);
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply test/format/ops.c:1134]: If we shift to an integer we'll be using a distribution 99% of the time which is a big shift in test format functionality, I don't think we want that do we? I could be misinterpreting what that integer represents and a 1 or 99 pareto is basically linear?

File: test/format/format_config_def.c [hunk 6]
```diff
@@ -2,33 +2,29 @@
 
 #include "format.h"
 
-CONFIG configuration_list[] = {
-  {"assert.read_timestamp", "assert read_timestamp",
-    C_BOOL, 2, 0, 0, V_GLOBAL_ASSERT_READ_TIMESTAMP},
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply test/format/format_config_def.c]: Yeah now that develop is merged this is a lot cleaner.

File: bench/wtperf/wtperf.c [hunk 7]
```diff
@@ -2889,22 +2888,9 @@ wtperf_rand(WTPERF_THREAD *thread)
...
-        U = 1 - (double)rval / (double)UINT32_MAX;
-        rval = (uint64_t)((pow(U, S1) - 1) * S2);
-        /*
-         * This Pareto calculation chooses out of range values about
-         * 2% of the time, from my testing. That will lead to the
-         * first item in the table being "hot".
-         */
-        if (rval > end_range)
-            rval = 0;
-    }
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply bench/wtperf/wtperf.c:2907]: I see what you mean, my plan is to standardize the function such that it returns a value between [0 and end) and the caller can choose to adapt the returned value accordingly.

File: test/format/ops.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (ddanderson) [reply test/format/ops.c]: Oh, right.  We still want no-pareto for most of the runs.  I still think it's useful to vary the skew parameter when we're running with pareto, we'll get a lot more variance.  So, how about we keep pareto as a bool and have a `pareto.skew` configuration that varies from 0 to 100?

[the remark this answers is not in the document]
Comment (luke-pearson) [reply test/format/ops.c]: Yeah I'll do that.

[the remark this answers is not in the document]
Comment (luke-pearson) [reply test/format/ops.c]: [2208b0c](https://github.com/wiredtiger/wiredtiger/pull/8961/commits/2208b0cf8046cf6b10685d1a9729f7d7d9116b33)

**Comment by luke-pearson** [comment]
Once this generally looks good I'll kick off stress test builds and perf builds to ensure no obvious change has come through.

**Review by sueloverso — APPROVED** [review]
LGTM - I think this is a good change overall.

**Review by ddanderson — APPROVED** [review]
LGTM!

**Review by ddanderson — APPROVED** [review]
LGTM

**Review by ddanderson — APPROVED** [review]
I don't really understand much of cmake, but this looks okay to me.

**Pull request merged**